test/e2e/parametrisable_templates/parametrisable_templates_test.go

```diff
@@ -0,0 +1,451 @@
+
+				harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+
+					(*device.Metadata.Labels)[revisionLabelKey] = branchTargetRevision
+					logrus.Infof("Updating the device with label %s=%s", revisionLabelKey, branchTargetRevision)
+				})
+
+				err = harness.WaitForDeviceNewGeneration(deviceId, nextGeneration)
+				Expect(err).ToNot(HaveOccurred())
+
+				gitConfigResponse, err = harness.GetDeviceGitConfig(device, gitConfigName)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(gitConfigResponse.GitRef.TargetRevision).To(ContainSubstring(revisionLabelValue))
+
+				By("Update the fleet inline config with a template variable with getOrDefault function")
+				nextRenderedVersion, err = harness.PrepareNextDeviceVersion(deviceId)
+				Expect(err).ToNot(HaveOccurred())
+
+				inlineConfigProviderSpec = v1alpha1.ConfigProviderSpec{}
+				err = inlineConfigProviderSpec.FromInlineConfigProviderSpec(inlineConfigValid)
+				Expect(err).ToNot(HaveOccurred())
+
+				deviceSpec.Config = &[]v1alpha1.ConfigProviderSpec{inlineConfigProviderSpec}
+
+				err = harness.CreateOrUpdateTestFleet(fleetTestName, testFleetSelector, deviceSpec)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = harness.WaitForDeviceNewRenderedVersion(deviceId, nextRenderedVersion)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Remove the team label from the device and check that the config variable is replaced by the default")
+				nextGeneration, err = harness.PrepareNextDeviceGeneration(deviceId)
+				Expect(err).ToNot(HaveOccurred())
+
+				logrus.Infof("Removing %s labels", teamLabelKey)
+				harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+
+					device.Metadata.Labels = &map[string]string{
+						fleetSelectorKey: fleetSelectorValue,
+						aliasKey:         deviceAlias,
+						configLabelKey:   configLabelValue,
+						revisionLabelKey: revisionLabelValue,
+						suffixLabelKey:   suffixLabelValue,
+					}
+					logrus.Infof("Updating %s with labels", deviceId)
+				})
+
+				By("Wait for the device to pick the fleet config")
+				err = harness.WaitForDeviceNewGeneration(deviceId, nextGeneration)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Check that the default variables are replaced in the config")
+				response, err = harness.Client.ReadDeviceWithResponse(harness.Context, deviceId)
+				Expect(err).ToNot(HaveOccurred())
+				device = response.JSON200
+				Expect(device).ToNot(BeNil(), "failed to read updated device")
+				logrus.Infof("The device labels are %s", *device.Metadata.Labels)
+
+				inlineConfigResponse, err = harness.GetDeviceInlineConfig(device, inlineConfigName)
+				Expect(err).ToNot(HaveOccurred())
+				if len(inlineConfigResponse.Inline) > 0 {
+					Expect(inlineConfigResponse.Inline[0].Path).To(ContainSubstring(defaultTeamLabelValue))
+					Expect(inlineConfigResponse.Inline[0].Content).To(ContainSubstring(defaultTeamLabelValue))
+				}
+
+			})
+	})
+})
+
+var (
+	fleetSelectorKey      = "fleet"
```

Comment:
I added resource types to constants
https://github.com/flightctl/flightctl/blob/main/test/util/constants.go#L6